src/views/AuthenticationWrapper/DevSecurity.tsx

```diff
@@ -43,45 +48,52 @@ const DevSecurity = ({ children }: ParentComponentProps) => {
   useEffect(() => {
     if (window.localStorage[storageKey]) {
       const state = JSON.parse(window.localStorage[storageKey]);
-      setAuthState(as => {
+      setAuthState(prevAuthState => {
         return {
-          ...as,
-          name: `User ${state.eua}`,
-          isAuthenticated: true,
-          euaId: state.eua,
-          groups: state.jobCodes
+          ...prevAuthState,
+          isAuthenticated: true
         };
       });
+      setEuaId(state.eua);
+      setGroups(state.jobCodes);
     }
   }, []);
 
-  const handleSubmit: ReactEventHandler = event => {
+  const handleSubmit = (event: React.FormEvent) => {
     event.preventDefault();
+    const activeGroups = jobCodes.filter(group => groups[group]);
     const value = {
       eua: euaId,
-      jobCodes: Array.from(checkboxValues.current)
+      jobCodes: activeGroups
     };
     localStorage.setItem(storageKey, JSON.stringify(value));
     localStorage.removeItem('okta-token-storage'); // ensure that the dev token is used
     setAuthState({
       ...authState,
-      isAuthenticated: true,
-      name: `User ${euaId}`,
-      euaId,
-      groups: Array.from(checkboxValues.current)
+      isAuthenticated: true
     });
   };
 
-  const checkboxChange: ReactEventHandler<HTMLInputElement> = event => {
-    if (event.currentTarget.checked) {
-      checkboxValues.current.add(event.currentTarget.value);
-    } else {
-      checkboxValues.current.delete(event.currentTarget.value);
-    }
+  const checkboxChange = (event: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
This feels a bit easier. We dont need to add a remove values from an array/set. We just set the value.

Additionally `groups` is it's own state. If it were in `authState` it'd look more complicated with more spreading;

e.g.
```
setAuthState(prevAuthState => {
  ...prevAuthState,
  groups: {
    ...prevAuthState.groups,
    [event.target.value]: event.target.checked
  }
}
```

Reply:
👍🏻  An object is a better data structure to use here as it more closely maps to how we're accessing the data.